File: mobile/src/main/java/com/windscribe/mobile/welcome/WelcomeView.java
/*
 * Copyright (c) 2021 Windscribe Limited.
 */

package com.windscribe.mobile.welcome;

import java.io.File;

public interface WelcomeView {

    void clearInputErrors();

    void goToSignUp();

    void gotoHomeActivity(boolean clearTop);

    void hideSoftKeyboard();

    void launchShareIntent(File file);

    void prepareUiForApiCallFinished();

    void prepareUiForApiCallStart();

    void setEmailError(String errorMessage);

    void setFaFieldsVisibility(int visible);

    void setLoginRegistrationError(String error);

    void setPasswordError(String error);

    void setTwoFaError(String errorMessage);

    void setUsernameError(String error);

    void showError(String error);

    void showFailedAlert(String error);

    void showNoEmailAttentionFragment(String username, String password, boolean accountClaim, boolean pro);

    void showToast(String message);

    void updateCurrentProcess(String mCurrentCall);
}

File: mobile/src/main/java/com/windscribe/mobile/welcome/WelcomeView.kt
/*
 * Copyright (c) 2021 Windscribe Limited.
 */
package com.windscribe.mobile.welcome

import java.io.File

interface WelcomeView {
    fun clearInputErrors()
    fun goToSignUp()
    fun gotoHomeActivity(clearTop: Boolean)
    fun hideSoftKeyboard()
    fun launchShareIntent(file: File)
    fun prepareUiForApiCallFinished()
    fun prepareUiForApiCallStart()
    fun setEmailError(errorMessage: String)
    fun setFaFieldsVisibility(visible: Int)
    fun setLoginRegistrationError(error: String)
    fun setPasswordError(error: String)
    fun setTwoFaError(errorMessage: String)
    fun setUsernameError(error: String)
    fun showError(error: String)
    fun showFailedAlert(error: String)
    fun showNoEmailAttentionFragment(
        username: String,
        password: String,
        accountClaim: Boolean,
        pro: Boolean
    )

    fun showToast(message: String)
    fun updateCurrentProcess(mCurrentCall: String)
}